### ISSUE TITLE
Way to dump the DOM and/or HTML on expect or test failure

### ISSUE THREAD

**Issue by TimGebhardt** [stirno/FluentAutomation#81]
When a test fails on our build slaves it's extremely helpful to get the DOM of the page at the time of failure.  Is there someway we could get this information similarly to how screenshots on failure are wired up?

**Comment by stirno** [comment]
Perhaps we can look at providing access to more data in the ExpectFailedCallback so that this could be implemented. I'll take a look soon.

**Comment by stirno** [comment]
v3.0 provides a WindowState object to the OnAssertFailed/OnExpectFailed event handlers so you can implement things like logging the state of the app when errors happened. This object, currently has the page URL and Source.

```
Config.OnAssertFailed((ex, state) => {
    // write page source to Trace
   Trace.WriteLine(state.Source);
});
```

**Comment by stirno** [comment]
Closing as this is included in v3.0. Pull it from fixmultibrowser branch (the current 3.0 branch) or wait for it to hit Nuget in a few days and let me know what you think! Thanks!
